src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs

```diff
@@ -1393,6 +1393,27 @@ public void LanguageVersion_TryParseDisplayString(string input, bool success, La
         }
 
         [Fact]
+        public void LanguageVersion_CommandLineUsage()
+        {
+            var expected = Enum.GetValues(typeof(LanguageVersion)).Cast<LanguageVersion>().Select(v => v.ToDisplayString());
+            string help = CSharpResources.IDS_CSCHelp;
+
+            var rangeStart = help.IndexOf("/langversion");
+            var rangeEnd = help.IndexOf("/delaysign");
+            Assert.True(rangeEnd > rangeStart);
+
+            string helpRange = help.Substring(rangeStart, rangeEnd - rangeStart).ToLowerInvariant();
+
+            foreach (var version in expected)
+            {
+                Assert.True(helpRange.Contains(version), $"Missing version '{version}'");
```

Comment:
Yes. Also, "1" matches "iso-1".
If we add "7.1" before "7", yes, we'll run into a problem. Also, if we add "7.8", then forget to add "8", we'll also run into a problem.

This check is not meant to remove the checklist, but to provide a reminder in common cases. I don't think it's worth it to fully parse the usage string. Let me know if you feel otherwise.